### PR TITLE
Update UI tests for Reader

### PR DESCRIPTION
**Pull request by dvdchr**
Refs #22463 

This updates the (previously disabled) Reader UI tests, along with some small fixes & improvements to identifiers for UI testing purposes. 

In the new version, we are moving away from "tabs" to "streams"; the biggest change is the stream switcher button. Within the UI test code, this replaces the previous `openDiscoverTab`, `openFollowingTab`, etc. to `switchToStream(_ stream: ReaderStream)`.

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
The UI tests were re-enabled.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
N/A. No user-facing UI changes in this PR.

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/UITestsFoundation/Screens/ReaderScreen.swift
Comment (tiagomar) [inline WordPress/UITestsFoundation/Screens/ReaderScreen.swift]:
```suggestion
        waitForExistenceAndTap(followButton, timeout: 3)
```

Hey you considered using [`waitForExistenceAndTap(_ element: XCUIElement, timeout: TimeInterval = 5)`](https://github.com/wordpress-mobile/WordPress-iOS/blob/76981b0569881be02ec1b3b9fb7f0288338d4a95/WordPress/UITestsFoundation/Globals.swift#L39)? It looks like a good case for that. =]

**Comment by tiagomar** [comment]
👋  Hey @dvdchr, the tests run without issues locally on iPhone and iPad. I'm waiting on [the CI to finish a re-run for iPad](https://buildkite.com/automattic/wordpress-ios/builds/19998#018d566a-5dcd-4ca8-857c-c18897fdbe00), but it looks like an issue is happening in CI.

The app seems to have crashed:

```

Incident Identifier: 5B41A21F-95FA-44E9-9606-3B31C3A29D5D
CrashReporter Key:   5FB0E8AE-EC84-4F76-BBE7-87CD33FE5F67
Hardware Model:      VirtualMac2,1
Process:             JetpackUITests-Runner [1902]
Path:                /Users/USER/Library/Developer/CoreSimulator/Devices/094F357F-9C42-4D5E-924F-0A75D1DE3897/data/Containers/Bundle/Application/2C59D5C3-1D90-44AD-B5F4-BC7EEBE04363/JetpackUITests-Runner.app/JetpackUITests-Runner
Identifier:          com.automattic.jetpack.JetpackUITests.xctrunner
Version:             1.0 (1)
Code Type:           ARM-64 (Native)
Role:                Non UI
Parent Process:      launchd_sim [1617]
Coalition:           com.apple.CoreSimulator.SimDevice.094F357F-9C42-4D5E-924F-0A75D1DE3897 [815]
Responsible Process: SimulatorTrampoline [887]

Date/Time:           2024-01-29 09:44:26.8642 -0800
Launch Time:         2024-01-29 09:23:29.4394 -0800
OS Version:          macOS 13.5.2 (22G91)
Release Type:        User
Report Version:      104

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000001, 0x00000001929f3938
Termination Reason: SIGNAL 5 Trace/BPT trap: 5
Terminating Process: exc handler [1902]

Triggered by Thread:  0

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libswiftCore.dylib            	       0x1929f3938 _assertionFailure(_:_:file:line:flags:) + 248
1   UITestsFoundation             	       0x10f7ddbc0 ReaderScreen.openLastPostComments() + 436
2   JetpackUITests                	       0x10bc0939c ReaderTests.testAddCommentToPost() + 136
3   JetpackUITests                	       0x10bc09538 @objc ReaderTests.testAddCommentToPost() + 48
4   CoreFoundation                	       0x180497750 0x180364000 + 1259344
5   CoreFoundation                	       0x180494ab4 0x180364000 + 1247924
6   XCTestCore                    	       0x1059c6d80 +[XCTFailableInvocation invokeErrorConventionInvocation:completion:] + 92
7   XCTestCore                    	       0x1059c6d1c __90+[XCTFailableInvocation invokeInvocation:withTestMethodConvention:lastObservedErrorIssue:]_block_invoke + 24
8   XCTestCore                    	       0x1059c67d8 __81+[XCTFailableInvocation invokeWithAsynchronousWait:lastObservedErrorIssue:block:]_block_invoke + 300
9   XCTestCore                    	       0x1059899e4 __49+[XCTSwiftErrorObservation observeErrorsInBlock:]_block_invoke + 28
10  libXCTestSwiftSupport.dylib   	       0x1058fb6a8 specialized TaskLocal.withValue<A>(_:operation:file:line:) + 104
11  libXCTestSwiftSupport.dylib   	       0x1058fb598 specialized static XCTSwiftErrorObservation._observeErrors(in:) + 360
12  libXCTestSwiftSupport.dylib   	       0x1058fb73c @objc static XCTSwiftErrorObservation._observeErrors(in:) + 48
13  XCTestCore                    	       0x1059898e0 +[XCTSwiftErrorObservation observeErrorsInBlock:] + 184
14  XCTestCore                    	       0x1059c65b8 +[XCTFailableInvocation invokeWithAsynchronousWait:lastObservedErrorIssue:block:] + 184
15  XCTestCore                    	       0x1059c6b94 +[XCTFailableInvocation invokeInvocation:withTestMethodConvention:lastObservedErrorIssue:] + 156
16  XCTestCore                    	       0x1059c702c +[XCTFailableInvocation invokeInvocation:lastObservedErrorIssue:] + 68
17  XCTestCore                    	       0x1059bacbc __24-[XCTestCase invokeTest]_block_invoke.78 + 96
18  XCTestCore                    	       0x105980ad4 -[XCTestCase(XCTIssueHandling) _caughtUnhandledDeveloperExceptionPermittingControlFlowInterruptions:caughtInterruptionException:whileExecutingBlock:] + 160
19  XCTestCore                    	       0x1059ba888 -[XCTestCase invokeTest] + 668
20  XCTestCore                    	       0x1059bc0d0 __26-[XCTestCase performTest:]_block_invoke.116 + 32
21  XCTestCore                    	       0x105980ad4 -[XCTestCase(XCTIssueHandling) _caughtUnhandledDeveloperExceptionPermittingControlFlowInterruptions:caughtInterruptionException:whileExecutingBlock:] + 160
22  XCTestCore                    	       0x1059bbb94 __26-[XCTestCase performTest:]_block_invoke.106 + 492
23  XCTestCore                    	       0x10599f244 +[XCTContext _runInChildOfContext:forTestCase:markAsReportingBase:block:] + 172
24  XCTestCore                    	       0x10599f144 +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 100
25  XCTestCore                    	       0x1059bb640 -[XCTestCase performTest:] + 248
26  XCTestCore                    	       0x10596ca38 -[XCTest runTest] + 44
27  XCTestCore                    	       0x1059a21bc -[XCTestSuite runTestBasedOnRepetitionPolicy:testRun:] + 64
28  XCTestCore                    	       0x1059a2088 __27-[XCTestSuite performTest:]_block_invoke + 160
29  XCTestCore                    	       0x1059a1b3c __59-[XCTestSuite _performProtectedSectionForTest:testSection:]_block_invoke + 40
30  XCTestCore                    	       0x10599f244 +[XCTContext _runInChildOfContext:forTestCase:markAsReportingBase:block:] + 172
31  XCTestCore                    	       0x10599f144 +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 100
32  XCTestCore                    	       0x1059a1ab8 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 152
33  XCTestCore                    	       0x1059a1d60 -[XCTestSuite performTest:] + 192
34  XCTestCore                    	       0x10596ca38 -[XCTest runTest] + 44
35  XCTestCore                    	       0x1059a21bc -[XCTestSuite runTestBasedOnRepetitionPolicy:testRun:] + 64
36  XCTestCore                    	       0x1059a2088 __27-[XCTestSuite performTest:]_block_invoke + 160
37  XCTestCore                    	       0x1059a1b3c __59-[XCTestSuite _performProtectedSectionForTest:testSection:]_block_invoke + 40
38  XCTestCore                    	       0x10599f244 +[XCTContext _runInChildOfContext:forTestCase:markAsReportingBase:block:] + 172
39  XCTestCore                    	       0x10599f144 +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 100
40  XCTestCore                    	       0x1059a1ab8 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 152
41  XCTestCore                    	       0x1059a1d60 -[XCTestSuite performTest:] + 192
42  XCTestCore                    	       0x10596ca38 -[XCTest runTest] + 44
43  XCTestCore                    	       0x1059a21bc -[XCTestSuite runTestBasedOnRepetitionPolicy:testRun:] + 64
44  XCTestCore                    	       0x1059a2088 __27-[XCTestSuite performTest:]_block_invoke + 160
45  XCTestCore                    	       0x1059a1b3c __59-[XCTestSuite _performProtectedSectionForTest:testSection:]_block_invoke + 40
46  XCTestCore                    	       0x10599f244 +[XCTContext _runInChildOfContext:forTestCase:markAsReportingBase:block:] + 172
47  XCTestCore                    	       0x10599f144 +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 100
48  XCTestCore                    	       0x1059a1ab8 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 152
49  XCTestCore                    	       0x1059a1d60 -[XCTestSuite performTest:] + 192
50  XCTestCore                    	       0x10596ca38 -[XCTest runTest] + 44
51  XCTestCore                    	       0x10596eb88 __89-[XCTTestRunSession executeTestsWithIdentifiers:skippingTestsWithIdentifiers:completion:]_block_invoke + 100
52  XCTestCore                    	       0x10599f244 +[XCTContext _runInChildOfContext:forTestCase:markAsReportingBase:block:] + 172
53  XCTestCore                    	       0x10599f144 +[XCTContext runInContextForTestCase:markAsReportingBase:block:] + 100
54  XCTestCore                    	       0x10596ea50 -[XCTTestRunSession executeTestsWithIdentifiers:skippingTestsWithIdentifiers:completion:] + 252
55  XCTestCore                    	       0x1059da44c __72-[XCTExecutionWorker enqueueTestIdentifiersToRun:testIdentifiersToSkip:]_block_invoke_2 + 108
56  XCTestCore                    	       0x1059da5ac -[XCTExecutionWorker runWithError:] + 124
57  XCTestCore                    	       0x10599bcec __25-[XCTestDriver _runTests]_block_invoke.183 + 52
58  XCTestCore                    	       0x105979094 -[XCTestObservationCenter _observeTestExecutionForTestBundle:inBlock:] + 180
59  XCTestCore                    	       0x10599b784 -[XCTestDriver _runTests] + 1024
60  XCTestCore                    	       0x10596d014 _XCTestMain + 88
61  JetpackUITests-Runner         	       0x104e61b38 _XCTRunnerRunTests + 100
62  JetpackUITests-Runner         	       0x104e61ad4 __46-[_XCTRunnerAppDelegate runTestsInBackground:]_block_invoke + 12
63  CoreFoundation                	       0x1803f1ce0 0x180364000 + 580832
64  CoreFoundation                	       0x1803f1428 0x180364000 + 578600
65  CoreFoundation                	       0x1803ebccc 0x180364000 + 556236
66  CoreFoundation                	       0x1803eb5a4 0x180364000 + 554404
67  GraphicsServices              	       0x18e9fbae4 GSEventRunModal + 160
68  UIKitCore                     	       0x1852f02e4 -[UIApplication _run] + 868
69  UIKitCore                     	       0x1852f3f5c UIApplicationMain + 124
70  JetpackUITests-Runner         	       0x104e61d98 main + 160
71  dyld_sim                      	       0x104eb9544 start_sim + 20
72  dyld                          	       0x104f99f28 0x104f94000 + 24360

Thread 1:: com.apple.uikit.eventfetch-thread
0   libsystem_kernel.dylib        	       0x1056991d0 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x1056aa298 mach_msg2_internal + 76
2   libsystem_kernel.dylib        	       0x1056a13d8 mach_msg_overwrite + 540
3   libsystem_kernel.dylib        	       0x105699540 mach_msg + 20
4   CoreFoundation                	       0x1803f1754 0x180364000 + 579412
5   CoreFoundation                	       0x1803ebe34 0x180364000 + 556596
6   CoreFoundation                	       0x1803eb5a4 0x180364000 + 554404
7   Foundation                    	       0x180dcd68c 0x180797000 + 6514316
8   Foundation                    	       0x180dcd8b0 0x180797000 + 6514864
9   UIKitCore                     	       0x185393b88 -[UIEventFetcher threadMain] + 404
10  Foundation                    	       0x180df3e70 0x180797000 + 6671984
11  libsystem_pthread.dylib       	       0x10577f428 _pthread_start + 116
12  libsystem_pthread.dylib       	       0x10577a648 thread_start + 8

Thread 2:: com.apple.NSURLConnectionLoader
0   libsystem_kernel.dylib        	       0x1056991d0 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x1056aa298 mach_msg2_internal + 76
2   libsystem_kernel.dylib        	       0x1056a13d8 mach_msg_overwrite + 540
3   libsystem_kernel.dylib        	       0x105699540 mach_msg + 20
4   CoreFoundation                	       0x1803f1754 0x180364000 + 579412
5   CoreFoundation                	       0x1803ebe34 0x180364000 + 556596
6   CoreFoundation                	       0x1803eb5a4 0x180364000 + 554404
7   CFNetwork                     	       0x184467f68 0x184250000 + 2195304
8   Foundation                    	       0x180df3e70 0x180797000 + 6671984
9   libsystem_pthread.dylib       	       0x10577f428 _pthread_start + 116
10  libsystem_pthread.dylib       	       0x10577a648 thread_start + 8

Thread 3:
0   libsystem_pthread.dylib       	       0x10577a634 start_wqthread + 0

Thread 4:
0   libsystem_pthread.dylib       	       0x10577a634 start_wqthread + 0

Thread 5:
0   libsystem_pthread.dylib       	       0x10577a634 start_wqthread + 0

Thread 6:
0   libsystem_pthread.dylib       	       0x10577a634 start_wqthread + 0

Thread 7:
0   libsystem_pthread.dylib       	       0x10577a634 start_wqthread + 0

Thread 8:
0   libsystem_pthread.dylib       	       0x10577a634 start_wqthread + 0


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x800000010f87ea20   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000600001790540
    x4: 0x0000000000000000   x5: 0x000000016af9e6f0   x6: 0x0000000000000064   x7: 0x0000000000000000
    x8: 0x0000000000000100   x9: 0x00000000000000ff  x10: 0x0000000000000500  x11: 0x00000000e7417051
   x12: 0x00000000000007fb  x13: 0x00000000000007fd  x14: 0x00000000e7617815  x15: 0x0000000000000015
   x16: 0x00000000e7417051  x17: 0x0000000000017800  x18: 0x0000000000000000  x19: 0x000000010f882a50
   x20: 0x800000010f87ea20  x21: 0x0000000000000000  x22: 0x000000000000000b  x23: 0x0000000000000024
   x24: 0x0000000000000073  x25: 0x000000010f87ea10  x26: 0xd00000000000002d  x27: 0x0000000000000000
   x28: 0x00000001f9cc3ea8   fp: 0x000000016af9e8d0   lr: 0x00000001929f3938
    sp: 0x000000016af9e800   pc: 0x00000001929f3938 cpsr: 0x60001000
   far: 0x0000000132a73408  esr: 0xf2000001 (Breakpoint) brk 1

Binary Images:
       0x104f94000 -        0x105023fff dyld (*) <e7a99595-e0f8-34af-be8b-9347d0d658a4> /usr/lib/dyld
       0x10bc00000 -        0x10bc63fff com.automattic.jetpack.JetpackUITests (1.0) <d5dc13a8-6956-3ddb-a509-ecd042a560e0> /Users/USER/Library/Developer/CoreSimulator/Devices/094F357F-9C42-4D5E-924F-0A75D1DE3897/data/Containers/Bundle/Application/2C59D5C3-1D90-44AD-B5F4-BC7EEBE04363/JetpackUITests-Runner.app/PlugIns/JetpackUITests.xctest/JetpackUITests
       0x10f778000 -        0x10f89bfff org.wordpress.UITestsFoundation (1.0) <f10b87a2-fa6d-3f91-858a-939119d7a1be> /opt/ci/*/UITestsFoundation.framework/UITestsFoundation
       0x1058e0000 -        0x105903fff libXCTestSwiftSupport.dylib (*) <e29152aa-16b1-366a-a706-a3befc542c97> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/lib/libXCTestSwiftSupport.dylib
       0x105340000 -        0x10534bfff libobjc-trampolines.dylib (*) <67a0ff9d-452e-3fe0-8b6e-ea95cf3a8b07> /Volumes/VOLUME/*/libobjc-trampolines.dylib
       0x105674000 -        0x105677fff com.apple.dt.XCTest (15.1) <e0d8b121-8bb5-3c16-9b2b-bcfafc8b904c> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/XCTest
       0x105968000 -        0x105a2bfff com.apple.dt.XCTestCore (15.1) <5546615e-eaa4-3696-9371-383ce506529d> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCTestCore.framework/XCTestCore
       0x104f84000 -        0x104f87fff com.apple.dt.XCUnit (15.1) <5c641ced-35bd-3aa0-a26a-cd56aa4d5e58> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCUnit.framework/XCUnit
       0x105ad4000 -        0x105b8bfff com.apple.dt.XCUIAutomation (15.1) <3467b1cb-ac6c-3385-a388-0a760690a7fc> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCUIAutomation.framework/XCUIAutomation
       0x1057f0000 -        0x105843fff com.apple.dt.XCTAutomationSupport (15.1) <4e12d457-5f00-326c-8357-cbaa777d241c> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework/XCTAutomationSupport
       0x1054fc000 -        0x1054fffff com.apple.dt.XCTestSupport (15.1) <7150fb44-7aac-338d-b0cd-761d9afd5aed> /Applications/Xcode-15.1.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/XCTestSupport.framework/XCTestSupport
       0x10572c000 -        0x105753fff com.apple.dt.instruments.DTXConnectionServices (15.1) <269bdec7-60d9-3600-80dd-643a232c18d6> /Volumes/VOLUME/*/DTXConnectionServices.framework/DTXConnectionServices
       0x105e4c000 -        0x105eebfff com.apple.dt.instruments.DVTInstrumentsFoundation (15.1) <eeced7f7-fee7-32c1-bc9d-ac22cca57e93> /Volumes/VOLUME/*/DVTInstrumentsFoundation.framework/DVTInstrumentsFoundation
       0x1054d4000 -        0x1054dbfff libsystem_platform.dylib (*) <9732e84b-ca21-3a30-9246-b6b230ce94fd> /usr/lib/system/libsystem_platform.dylib
       0x105698000 -        0x1056cffff libsystem_kernel.dylib (*) <507d31b7-7c2c-38ce-b864-0bae4ad4034b> /usr/lib/system/libsystem_kernel.dylib
       0x105778000 -        0x105787fff libsystem_pthread.dylib (*) <27ce1820-cc5b-30ea-974a-1d844310c438> /usr/lib/system/libsystem_pthread.dylib
       0x1057b4000 -        0x1057cffff com.apple.mlcompiler.services (95) <25f3a985-ceb4-3de6-be6b-d78298983f92> /Volumes/VOLUME/*/MLCompilerServices.framework/MLCompilerServices
       0x105520000 -        0x105563fff libMainThreadChecker.dylib (*) <4d4714d7-b2ad-3693-8f0a-9b2d6184b350> /Volumes/VOLUME/*/libMainThreadChecker.dylib
       0x104eb8000 -        0x104efffff dyld_sim (*) <45b5018b-7ae1-3c95-86a5-d89fd6cb36ae> /Volumes/VOLUME/*/dyld_sim
       0x104e5c000 -        0x104e63fff com.automattic.jetpack.JetpackUITests.xctrunner (1.0) <37c1f90a-a7b1-3272-9240-c18076d09522> /Users/USER/Library/Developer/CoreSimulator/Devices/094F357F-9C42-4D5E-924F-0A75D1DE3897/data/Containers/Bundle/Application/2C59D5C3-1D90-44AD-B5F4-BC7EEBE04363/JetpackUITests-Runner.app/JetpackUITests-Runner
       0x1929c5000 -        0x192e2efff libswiftCore.dylib (*) <d6a5eacd-c6ad-353c-bce9-2174692535d4> /Volumes/VOLUME/*/libswiftCore.dylib
       0x180364000 -        0x180717fff com.apple.CoreFoundation (6.9) <fc74142f-1b8b-3994-9466-f5061dd43152> /Volumes/VOLUME/*/CoreFoundation.framework/CoreFoundation
       0x18e9f8000 -        0x18ea00fff com.apple.GraphicsServices (1.0) <7cea06b5-77ce-3388-bb85-9a6942ca7f99> /Volumes/VOLUME/*/GraphicsServices.framework/GraphicsServices
       0x18471f000 -        0x186085fff com.apple.UIKitCore (1.0) <d0dea36b-57cc-3509-be3f-07785430decd> /Volumes/VOLUME/*/UIKitCore.framework/UIKitCore
       0x180797000 -        0x18123dfff com.apple.Foundation (6.9) <03b66a9f-be07-39ef-812d-ea7d8b87f7eb> /Volumes/VOLUME/*/Foundation.framework/Foundation
       0x184250000 -        0x1845e1fff com.apple.CFNetwork (1.0) <12886c7c-4c78-3c51-82a9-87ea2bff7b7d> /Volumes/VOLUME/*/CFNetwork.framework/CFNetwork

EOF


```

File: WordPress/UITestsFoundation/Screens/ReaderScreen.swift
Comment (dvdchr) [reply WordPress/UITestsFoundation/Screens/ReaderScreen.swift]: Thank you! Fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/22469/commits/ead4d8f1f2a7c97d5153a0339b4a6f7455aac71a.

**Comment by dvdchr** [comment]
👋🏼  @tiagomar, thanks for taking a look! Yeah, I was a bit stumped by this. Even though these are passing locally for iPhone and iPad simulators, they are failing on the CI. I downloaded the `.xctestresults` file, and it looks like the instruments are running properly... But here is the error for a failing `testViewPost()` case:

Screenshot 2024-01-30 at 15 41 52

So it looks like the UI test instrument tries to find a cell located at index: 9, but the error states that there are only 3 cells. This points out the possibility of a race condition for the `getLastPost()` method, where we are fetching the list of cells while the screen is still in its loading state.

Here is where we configure the Ghost loading animations, and we can see that we are displaying 10 rows for the Reader which explains where the index 9 is coming from.

https://github.com/wordpress-mobile/WordPress-iOS/blob/b1ba26954d94b1052143c4bebab9425f7f1a4ebb/WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController%2BGhost.swift#L25

To fix this, in https://github.com/wordpress-mobile/WordPress-iOS/pull/22469/commits/da1a495f14b74435a939afda35aadc024e1eb590 I've added an extra step for the test case to wait for the ghost loading animation to complete before proceeding to `getLastPost()`. Hopefully this will work!

**Comment by dvdchr** [comment]
Okay, from the [latest CI run](https://buildkite.com/automattic/wordpress-ios/builds/20027#018d59ca-1137-48c9-8531-459dea85c128), it seems the issue with `testViewPost` and `testViewPostInSafari` is solved 🎉 . However, there's still an error with the Follow Topic test case that somehow only occurs on the CI. Will download the `xctestresults` and see what's wrong with it.

**Comment by dvdchr** [comment]
It finally passed after a retry! Based on the previous failing `xctestresult`, it seems that the tap query is already targeting the correct table cell:

```
Failed to failed to compute hit point for Cell, {{287.5, 128.0}, {70.2, 33.5}}, identifier: 'topics-card-cell-button', label: 'Travel': Activation point invalid and no suggested hit points based on element frame. (Underlying Error: Activation point invalid and no suggested hit points based on element frame)
```

But looking closer at the UI hierarchy, the cell *does* exist! 🤯 I'm not sure why this doesn't work sometimes. Perhaps XCUITest's `tap` method is buggy when used on table cells?

```
Cell, 0x105926920, {{287.5, 128.0}, {70.2, 33.5}}, identifier: 'topics-card-cell-button', label: 'Travel' <AX element pid: 2527, elementOrHash.elementID: 105553157032672.3585>
  Other, 0x105926a40, {{287.5, 128.0}, {70.2, 33.5}} <AX element pid: 2527, elementOrHash.elementID: 4500244768.2316>
    StaticText, 0x105926b60, {{303.5, 136.0}, {38.0, 17.5}}, label: 'Travel' <AX element pid: 2527, elementOrHash.elementID: 4500459888.2548>
Cell, 0x105926c80, {{365.7, 128.0}, {76.2, 33.5}}, identifier: 'topics-card-cell-button', label: 'Photos' <AX element pid: 2527, elementOrHash.elementID: 105553157040832.3586>
  Other, 0x105926da0, {{365.7, 128.0}, {76.2, 33.5}} <AX element pid: 2527, elementOrHash.elementID: 4500025072.2317>
    StaticText, 0x105926ec0, {{381.7, 136.0}, {44.0, 17.5}}, label: 'Photos' <AX element pid: 2527, elementOrHash.elementID: 4500153104.2550>
Cell, 0x105926fe0, {{449.9, 128.0}, {103.8, 33.5}}, identifier: 'topics-card-cell-button', label: 'Technology' <AX element pid: 2527, elementOrHash.elementID: 105553157044864.3587>
  Other, 0x105927100, {{449.9, 128.0}, {103.8, 33.5}} <AX element pid: 2527, elementOrHash.elementID: 4500090432.2318>
    StaticText, 0x105927220, {{465.9, 136.0}, {72.0, 17.5}}, label: 'Technology' <AX element pid: 2527, elementOrHash.elementID: 4500337040.2552>
StaticText, 0x105927340, {{192.0, 100.0}, {640.0, 16.0}}, label: 'You might like' <AX element pid: 2527, elementOrHash.elementID: 4499828976.1900>
Cell, 0x105927460, {{192.0, 128.0}, {87.5, 33.5}}, identifier: 'topics-card-cell-button', label: 'Blogging' <AX element pid: 2527, elementOrHash.elementID: 105553157043328.3588>
  Other, 0x105927580, {{192.0, 128.0}, {87.5, 33.5}} <AX element pid: 2527, elementOrHash.elementID: 4499965008.2319>
    StaticText, 0x1059276a0, {{208.0, 136.0}, {55.5, 17.5}}, label: 'Blogging' <AX element pid: 2527, elementOrHash.elementID: 4499819328.2554>
Other, 0x1059277c0, {{8.0, 225.0}, {452.0, 1.0}} <AX element pid: 2527, elementOrHash.elementID: 4499732192.1901>
```

**Comment by tiagomar** [comment]
>  > Failed to failed to compute hit point for Cell, {{287.5, 128.0}, {70.2, 33.5}}, identifier: 'topics-card-cell-button', label: 'Travel': Activation point invalid and no suggested hit points based on element frame. (Underlying Error: Activation point invalid and no suggested hit points based on element frame)

> But looking closer at the UI hierarchy, the cell does exist! 🤯 I'm not sure why this doesn't work sometimes. Perhaps XCUITest's tap method is buggy when used on table cells?

Yeah, I've seen this happening several times before. From my previous investigations, it seems to be a timing issue associated with high CPU load. While it's not rare to observe it in CI, it's really difficult to reproduce it in our laptops. I've only managed to reproduce it locally with an induced high CPU load by running tests in parallel (4 simulators) + video editing software 😅. My hypothesis is based on an assumption: the elements are created as `{{0,0}, {0,0}}` and then inflated, in this case to `{{287.5, 128.0}, {70.2, 33.5}}`. I didn't confirm this from the Simulator specs (don't even know if such thing is available), but I've seen cases where the error messages did display `{{0,0}, {0,0}}` as element dimensions. So, I think that, due to simulator slowness, the automation might find `{{0,0}, {0,0}}` elements and breaks trying to calculate `isHittable`.

**Review by tiagomar — APPROVED** [review]
Thanks for incorporating the suggestion, @dvdchr !

Now that the iPad issue running in CI is fixed and all tests are passing, this is ready to go. 👍

**Comment by dvdchr** [comment]
> From my previous investigations, it seems to be a timing issue associated with high CPU load. While it's not rare to observe it in CI, it's really difficult to reproduce it in our laptops. I've only managed to reproduce it locally with an induced high CPU load by running tests in parallel (4 simulators) + video editing software 😅.

I see. That makes sense! It just feels odd to me that the action failed 3 times consecutively. Perhaps the query was run during an invalid or high CPU load state, and that caused the UI hierarchy (along with the coordinate information) to get corrupted. 🤔

In any case, thank you for reviewing and investigating the failures @tiagomar! 🙂 Just a note, this PR targets the feature branch `feature/reader-navigation-filter`. Once we're ready to merge to `trunk`, the UI testing group may be pinged again in that PR. But it should be just be requiring an approval from the team as it will contain no new changes for the UI test.

**Pull request merged**
